datalake/src/main/java/com/sequenceiq/datalake/flow/dr/restore/DatalakeRestoreActions.java

```diff
@@ -65,6 +69,46 @@
     @Inject
     private SdxService sdxService;
 
+    @Bean(name = "DATALAKE_TRIGGERING_RESTORE_STATE")
+    public Action<?, ?> triggerDatalakeRestore() {
+        return new AbstractSdxAction<>(DatalakeTriggerRestoreEvent.class) {
+            @Override
+            protected SdxContext createFlowContext(FlowParameters flowParameters, StateContext<FlowState, FlowEvent> stateContext,
+                    DatalakeTriggerRestoreEvent payload) {
+                return SdxContext.from(flowParameters, payload);
+            }
+
+            @Override
+            protected void prepareExecution(DatalakeTriggerRestoreEvent payload, Map<Object, Object> variables) {
+                variables.put(OPERATION_ID, payload.getDrStatus().getOperationId());
+                super.prepareExecution(payload, variables);
+            }
+
+            @Override
+            protected void doExecute(SdxContext context, DatalakeTriggerRestoreEvent payload, Map<Object, Object> variables) {
+                LOGGER.info("Triggering datalake restore for {}", payload.getResourceId());
+                DatalakeDrStatusResponse restoreStatusResponse =
+                        sdxBackupRestoreService.triggerDatalakeRestore(payload.getResourceId(),
+                                payload.getBackupId(),
+                                payload.getBackupLocationOverride(),
+                                payload.getUserId());
+                variables.put(RESTORE_ID, restoreStatusResponse.getDrOperationId());
+                variables.put(OPERATION_ID, restoreStatusResponse.getDrOperationId());
```

Comment:
Why are two variables (RESTORE_ID and OPERATION_ID) needed with the same value?

Reply:
Yes. They are the same after we get the restoreID from the thunderhead service. We usually use operation_id for any operation we do and restore_id for the id in the dr table. Here they are the same. Let's just keep both. There is no harm and have both is more clear simantically.